### PR TITLE
Fixed deserialization of objects of a generic type with generic arguments

### PR DESCRIPTION
**Pull request by Dieler**
…arguments

### CODE REVIEW

File: javers-core/src/test/groovy/org/javers/core/metamodel/type/MapTypeTest.groovy [hunk 1]
```diff
@@ -63,4 +66,17 @@ class MapTypeTest extends Specification{
         mType.keyType == String
         mType.valueType == Integer
     }
+
+    def "should return key & value Class if baseJavaType and value type are generic"(){
```
Comment (bartoszwalacik) [inline javers-core/src/test/groovy/org/javers/core/metamodel/type/MapTypeTest.groovy]: this new test is passing also on master so we are missing something here...

Comment (Dieler) [inline javers-core/src/test/groovy/org/javers/core/metamodel/type/MapTypeTest.groovy]: Ok, MapTypeTest is successful but TypeMapperTest fails without the fix in ParametrizedDehydratedType. Anyway, I've modified SnapshotRepositry and JaversRepositoryE2ETest to highlight the actual problem.

Comment (bartoszwalacik) [inline javers-core/src/test/groovy/org/javers/core/metamodel/type/MapTypeTest.groovy]: thanks, you are right, `mapper.getDehydratedType()` should be "recursive"

File: javers-core/src/main/java/org/javers/core/metamodel/type/ParametrizedDehydratedType.java [hunk 2]
```diff
@@ -19,7 +19,11 @@
         this.actualTypeArguments =
                 Lists.transform(javersType.getActualTypeArguments(), new Function<Type, Type>() {
                     public Type apply(Type input) {
-                        return typeMapper.getJaversType(input).getRawDehydratedType();
+                        JaversType javersType = typeMapper.getJaversType(input);
+                        if (!javersType.isGenericType()){
+                            return javersType.getRawDehydratedType();
+                        }
+                        return new ParametrizedDehydratedType(javersType, typeMapper);
```
Comment (bartoszwalacik) [inline javers-core/src/main/java/org/javers/core/metamodel/type/ParametrizedDehydratedType.java]: All I need to do is to manage the code repetition (with `typeMapper.getDehydratedType()`)

**Pull request merged**